### PR TITLE
feat: data resource details page

### PR DESCRIPTION
**Pull request by debjyoti-pandit**
Data resource details page.

image

Note: delete is not functioning properly as after deleting the resource, the UI is getting stuck and needs refreshing

### CODE REVIEW

**Comment by cloudbehl** [comment]
Is it a still a draft PR?

**Comment by debjyoti-pandit** [comment]
@cloudbehl blocked with some router issue, I can move the draft stage and proceed with review. but in this PR deleting data resource is deleting the resource but failing to navigate back to proper UI and crashing.

File: yarn.lock [hunk 1]
```diff
@@ -793,6 +793,11 @@
   resolved "https://registry.yarnpkg.com/@patternfly/react-icons/-/react-icons-4.43.15.tgz#8dd696ba92867b1074eee6ef7c672833197f6046"
   integrity sha512-b1fcPpIsQVJBGG1onpNTDiGqVJ60BVNy3qvzUu4Ea+lOwP83oIlhODtS792wx1B284a4wxP3uMc6uXCfdULD/w==
 
+"@patternfly/react-icons@^4.49.5":
+  version "4.49.5"
+  resolved "https://registry.yarnpkg.com/@patternfly/react-icons/-/react-icons-4.49.5.tgz#ac8c0282b904e8016a0a1dd4db63988e121b4358"
+  integrity sha512-pR04R32KZAd4uPWeFf/pYXp0uUTwDotsM1iPbjJ6Rpl1sELM4LuQV4gaG22dWLy24VCRizFoP6BSzDxmq+GY4g==
+
```
Comment (SanjalKatiyar) [inline yarn.lock:800]: why yarn file is updated ?????
when u haven't changed any dependency in the package.json... plz take a look... it is important...

File: packages/components/resource-pages/data-resource-list-view.tsx [hunk 2]
```diff
@@ -129,11 +130,19 @@ const RowRenderer: React.FC<RowProps<NamespaceStoreKind, CustomData>> = ({
 }) => {
   const { t } = useTranslation();
   const { launchModal, kebabActions, bucketPolicyMap } = rowData;
-  const bucketsCount = bucketPolicyMap[obj?.metadata?.name] || 0;
+  const nameSpaceName = getName(obj);
```
Comment (SanjalKatiyar) [inline packages/components/resource-pages/data-resource-list-view.tsx:133]:
```suggestion
  const dataResourceName = getName(obj);
```
we decided all variables will be data-resource and text will be data-source

File: packages/components/resource-pages/NamespaceStoreDetailsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import { Kebab } from '../../utils/kebab/kebab';
+import { useModalLauncher } from '../../utils/modals/modalLauncher';
+import { CommonDetails } from './CommonDetails';
+import ProviderDetails from './Providers';
+
+type DetailsProps = {
+  obj: NamespaceStoreKind;
+};
+
+type DetailsType = (launchModal: any, t) => React.FC<DetailsProps>;
```
Comment (SanjalKatiyar) [inline packages/components/resource-pages/NamespaceStoreDetailsPage.tsx:21]: both of these have proper types... plz check and add those...

File: packages/components/resource-pages/NamespaceStoreDetailsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    },
+    {
+      name: t('NamespaceStores'),
+      path: '/mcgms/cluster/resource/noobaa.io~v1alpha1~NamespaceStore',
+    },
+    {
+      name: t('NamespaceStore details'),
+      path: '',
+    },
+  ];
```
Comment (SanjalKatiyar) [inline packages/components/resource-pages/NamespaceStoreDetailsPage.tsx:76]:
```suggestion
  const breadcrumbs = [
    {
      name: t('Data Federation'),
      path: '/mcgms/cluster',
    },
    {
      name: t('Data source'),
      path: '/mcgms/cluster/resource/noobaa.io~v1alpha1~NamespaceStore',
    },
    {
      name: t('Data source details'),
      path: '',
    },
  ];
```

File: packages/utils/modals/DeleteModal.tsx [hunk 5]
```diff
@@ -102,7 +104,7 @@ const DeleteModal: React.FC<CommonModalProps<DeleteModalExtraProps>> = ({
         // If we are currently on the deleted resource's page, redirect to the resource list page
         const re = new RegExp(`/${resource.metadata.name}(/|$)`);
         if (re.test(window.location.pathname)) {
-          history.replaceState('/', '');
+          history.goBack();
```
Comment (SanjalKatiyar) [inline packages/utils/modals/DeleteModal.tsx:107]: what will happen when we delete from list page then ??
we don't want to `goBack` then... how is it working in odf-console ??

File: packages/components/resource-pages/data-resource-list-view.tsx (hunk 2 above)
Comment (debjyoti-pandit) [reply packages/components/resource-pages/data-resource-list-view.tsx:133]: will do, it once all the review comment is fixed.

wanted to have the single details page related commit first then second commit for all name change

File: packages/utils/modals/DeleteModal.tsx (hunk 5 above)
Comment (debjyoti-pandit) [reply packages/utils/modals/DeleteModal.tsx:107]: had the same code in ODF but it is behaving somewhat different

File: packages/utils/common.ts [hunk 6]
```diff
@@ -5,9 +5,13 @@ import {
   K8sResourceKindReference,
 } from '@openshift-console/dynamic-plugin-sdk';
 import { K8sModel } from '@openshift-console/dynamic-plugin-sdk/lib/api/common-types';
+import { GetAPIVersionForModel } from '@openshift-console/dynamic-plugin-sdk/lib/api/k8s-types';
 
 export const LAST_LANGUAGE_LOCAL_STORAGE_KEY = 'bridge/last-language';
 
+export const getAPIVersionForModel: GetAPIVersionForModel = (model) =>
+  !model?.apiGroup ? model.apiVersion : `${model.apiGroup}/${model.apiVersion}`;
+
```
Comment (SanjalKatiyar) [inline packages/utils/common.ts:14]: already exposed by sdk... plz check once...

File: yarn.lock (hunk 1 above)
Comment (debjyoti-pandit) [reply yarn.lock:800]: I had a merge conflict and let yarn resolve the dependencies on its own. Seems it is the culprit.

File: packages/utils/Conditions.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+  );
+};
+Conditions.displayName = 'Conditions';
+
+export type ConditionsProps = {
+  conditions: K8sResourceCondition[] | ClusterServiceVersionCondition[];
+  title?: string;
+  subTitle?: string;
+  type?: keyof typeof ConditionTypes;
+};
```
Comment (SanjalKatiyar) [inline packages/utils/Conditions.tsx:138]: add this file under `utils/details-page`

File: packages/utils/providers.ts [hunk 8]
```diff
@@ -43,3 +44,22 @@ export const secretPayloadCreator = (
...
+  });
+  return type;
+};
+
+export const getRegion = (
+  bs: BackingStoreKind | NamespaceStoreKind
+): string => {
+  const type = getMCGStoreType(bs);
+  return bs.spec?.[PROVIDERS_NOOBAA_MAP[type]]?.region;
+};
```
Comment (SanjalKatiyar) [inline packages/utils/providers.ts:65]: this should go under noobaa-utils IMO...

File: yarn.lock (hunk 1 above)
Comment (SanjalKatiyar) [reply yarn.lock:800]: this is important...

File: packages/utils/modals/DeleteModal.tsx (hunk 5 above)
Comment (SanjalKatiyar) [reply packages/utils/modals/DeleteModal.tsx:107]: this is also important...

Comment (debjyoti-pandit) [reply packages/utils/modals/DeleteModal.tsx:107]: parked it for now, created https://issues.redhat.com/browse/DFMS-123

File: packages/types/noobaa.ts [hunk 9]
```diff
@@ -1,15 +1,19 @@
...
     [key in nsSpecProvider]: {
-      [key: string]: string;
+      [key: string]: string | nsSecretObject;
     };
   } & {
     type: nsSpecType;
   };
+  status: {
+    conditions: K8sResourceCondition[];
+  };
```
Comment (SanjalKatiyar) [inline packages/types/noobaa.ts:16]: ToDo... not blocking the PR...
```suggestion
  status?: {
    conditions?: K8sResourceCondition[];
  };
```

**Pull request merged**